File: src/org/genyris/datetime/AbstractDateTimeFunction.java
// Copyright 2008 Peter William Birch <birchb@genyis.org>
//
// This software may be used and distributed according to the terms
// of the Genyris License, in the file "LICENSE", incorporated herein by reference.
//
package org.genyris.datetime;

import org.genyris.exception.GenyrisException;
import org.genyris.interp.ApplicableFunction;
import org.genyris.interp.Interpreter;
import org.genyris.interp.UnboundException;

public abstract class AbstractDateTimeFunction extends ApplicableFunction {


    public AbstractDateTimeFunction(Interpreter interp,
            String name) {
        super(interp, name, true);
    }

    public static void bindFunctionsAndMethods(Interpreter interpreter) throws UnboundException, GenyrisException {
        interpreter.bindGlobalProcedureInstance(new FormatDateFunction(interpreter));
        interpreter.bindGlobalProcedureInstance(new DetailedDateTimeFunction(interpreter));

    }

}

File: src/org/genyris/datetime/AbstractDateTimeFunction.kt
// Copyright 2008 Peter William Birch <birchb@genyis.org>
//
// This software may be used and distributed according to the terms
// of the Genyris License, in the file "LICENSE", incorporated herein by reference.
//
package org.genyris.datetime

import org.genyris.core.Constants
import org.genyris.core.PrefixSymbol
import org.genyris.exception.GenyrisException
import org.genyris.interp.ApplicableFunction
import org.genyris.interp.Interpreter

abstract class AbstractDateTimeFunction(
    interp: Interpreter,
    name: String?
) : ApplicableFunction(interp, PrefixSymbol(Constants.GENYRIS + "date#", name, name), true) {
    companion object {
        @kotlin.Throws(GenyrisException::class)
        fun bindFunctionsAndMethods(interpreter: Interpreter) {
            interpreter.bindGlobalProcedureInstance(FormatDateFunction(interpreter))
            interpreter.bindGlobalProcedureInstance(DetailedDateTimeFunction(interpreter))
        }
    }
}